src/js/utils/i18n.ts

```diff
@@ -1,31 +1,37 @@
 import { En, TranslateDictionary, TranslateKeys } from '../lang/en.js';
 
-const translationsLanguages = {
-  en: En,
+const translations: Record<string, TranslateDictionary> = {
+  en: En
 };
 
-let currentLanguage = globalThis.navigator?.language?.split('-')[0] || 'en';
+let currentLang = globalThis.navigator?.language || 'en';
 
 export const setLanguage = (langCode: string) => {
-  currentLanguage = langCode;
+  currentLang = langCode;
 };
 
 export const addTranslation = (
-  langCode: string,
+  lang: string,
   languageDictionary: TranslateDictionary
 ) => {
-  translationsLanguages[langCode] = languageDictionary;
+  translations[lang] = languageDictionary;
 };
 
-export const t = (
-  key: TranslateKeys,
-  variables: Record<string, string | number> = {}
-) => {
-  const result = translationsLanguages[currentLanguage]?.[key] || En[key];
+const resolveTranslation = (key: TranslateKeys): string => {
+  const [base] = currentLang.split('-');
 
-  return result.replace(/\{(\w+)\}/g, (_, varName) =>
-    variables[varName] !== undefined
-      ? String(variables[varName])
-      : `{${varName}}`
+  return (
+    translations[currentLang]?.[key] ||
+    translations[base]?.[key] ||
+    translations.en?.[key] ||
+    key
   );
 };
+
+export const t = (
+  key: TranslateKeys,
+  vars: Record<string, string | number> = {}
+): string =>
+  resolveTranslation(key).replace(/\{(\w+)\}/g, (_, v) =>
+    v in vars ? String(vars[v]) : `{${v}}`
+  );
```

Comment:
```suggestion
  );

```

always end with an empty newline. should be a text editor setting that is done when saving